plugins/modules/aws_api_gateway_domain.py

```diff
@@ -0,0 +1,333 @@
+    default: EDGE
+    choices: ['EDGE', 'REGIONAL', 'PRIVATE']
+    type: str
+  domain_mappings:
+    description:
+      - Map your domain base paths to your API GW REST APIs, you previously created. Use provide ID of the API setup and the release stage.
+      - "domain_mappings should be a list of dictionaries containing three keys: base_path, rest_api_id and stage."
+      - "Example: I([{ base_path: v1, rest_api_id: abc123, stage: production }])"
+      - if you want base path to be just I(/) omit the param completely or set it to empty string.
+    required: true
+    type: list
+  state:
+    description:
+      - Create or delete custom domain setup
+    default: present
+    choices: [ 'present', 'absent' ]
+    type: str
+extends_documentation_fragment:
+    - aws
+    - ec2
+notes:
+   - Does not create a DNS entry on Route53, for that use the route53 module.
+   - Only supports TLS certificates from AWS ACM that can just be referenced by the ARN, while the AWS API still offers (deprecated)
+     options to add own Certificates.
+'''
+
+EXAMPLES = '''
+- name: Setup endpoint for a custom domain for your API Gateway HTTP API
+  aws_api_gateway_domain:
+    domain_name: myapi.foobar.com
+    certificate_arn: 'arn:aws:acm:us-east-1:1231123123:certificate/8bd89412-abc123-xxxxx'
+    security_policy: TLS_1_0
+    endpoint_type: EDGE
+    domain_mappings:
+        - { rest_api_id: abc123, stage: production }
+    state: present
+  register: api_gw_domain_result
+
+- name: Create a DNS record for your custom domain on route 53 (using route53 module)
+  route53:
+    record: myapi.foobar.com
+    value: "{{ api_gw_domain_result.response.domain.distribution_domain_name }}"
+    type: A
+    alias: true
+    zone: foobar.com
+    alias_hosted_zone_id: "{{ api_gw_domain_result.response.domain.distribution_hosted_zone_id }}"
+    command: create
+'''
+
+RETURN = '''
+response:
+  description: The data returned by create_domain_name (or update and delete) nad create_base_path_mapping methods in boto3
+  returned: success
+  type: dict
+  sample:
+    domain:
+        {
+            domain_name: mydomain.com,
+            certificate_arn: 'arn:aws:acm:xxxxxx',
+            distribution_domain_name: xxxx.cloudfront.net,
+            distribution_hosted_zone_id: ABC123123,
+            endpoint_configuration: { types: ['EDGE'] },
+            domain_name_status: 'AVAILABLE',
+            security_policy: TLS_1_2,
+            tags: {}
+        }
+    path_mappings: [
+        { base_path: '(empty)', rest_api_id: 'abcd123', stage: 'production' }
+    ]
+'''
+
+try:
+    import botocore
+except ImportError:
+    # HAS_BOTOCORE taken care of in AnsibleAWSModule
+    pass
+
+import copy
+import traceback
+
+from ansible.module_utils.aws.core import AnsibleAWSModule, is_boto3_error_code
+from ansible.module_utils.ec2 import (AWSRetry, camel_dict_to_snake_dict, snake_dict_to_camel_dict)
```

Comment:
Build fails on imports now, as those modules seem to be not available anymore in this project. Is there a new way to include them now?